koza/model/source.py

```diff
@@ -98,4 +100,5 @@ def _get_row(self):
                 row = next(self._reader)
         else:
             row = next(self._reader)
+        self.last_row = row
```

Comment:
curious why this is needed? maybe some comments would help here

Reply:
Oh, good call, that's in there to print the row that won't validate alongside the validation exception, but that's definitely not self explanatory from the code.